app/core/models/abortables.ts

```diff
@@ -0,0 +1,54 @@
+import {nanoid} from "@reduxjs/toolkit"
+import {isString, remove} from "lodash"
+
+export type Abortable<Meta extends any = any> = {
+  id: string
+  tab?: string
+  tag?: string
+  meta?: Meta
+  abort: () => void | Promise<void>
+}
+
+type NewAbortable = Omit<Abortable, "id">
+
+export class Abortables {
```

Comment:
This interface seems great, I like the predicate options. My only hesitancy is that it might read more clearly and be a little safer if we have separate and explicit methods for the most destructive abort/remove all options (e.g. `.abort` would require it's predicate instead of being optional, and then there would be a separate `.abortAll` method).

Reply:
Agreed! I'll add abort all. And make abort require a predicate.